redbot/core/utils/__init__.py

```diff
@@ -255,3 +256,54 @@ def bounded_gather(
     tasks = (_sem_wrapper(semaphore, task) for task in coros_or_futures)
 
     return asyncio.gather(*tasks, return_exceptions=return_exceptions)
+
+
+class AsyncGen(AsyncIterable[_T]):
+    """Yield entry every `delay` seconds per `steps`."""
+
+    def __init__(
+        self, contents: Union[Iterator[_T], Iterable[_T]], delay: float = 0.0, steps: int = 1
+    ) -> None:
+        self.delay = delay
+        self.content = iter(contents)
+        self.i = 0
+        self.steps = steps
+
+    def __aiter__(self):
+        return self
+
+    async def __anext__(self) -> Awaitable[_T]:
```

Comment:
Incorrect type hint, it should be:
```suggestion
    async def __anext__(self) -> _T:
```